TestRunner/Program.cs

```diff
@@ -44,45 +44,47 @@ static void Main(string[] args)
         {
             LoadAllTestAssemblies();
 
-            if (args.Length == 0)
+
+            while (args.Length == 0 || !m_TestMethods.ContainsKey(args[0]))
             {
-                Console.WriteLine("Please provide a filter for the methods you want to run. This can be either the name of the method or its namespace (after 'BH.Test')");
-                return;
+                if (args.Length == 0)
+                    Console.WriteLine("Please provide a filter for the methods you want to run. This can be either the name of the method or its namespace (after 'BH.Test')");
+                else
+                    Console.WriteLine("Cannot find any test matching " + args[0]);
```

Comment:
I would make sure the instruction is displayed regardless of whether the input is not provided or not found

```suggestion
                if (args.Length != 0)
                    Console.WriteLine("Cannot find any test matching " + args[0]);
                
                Console.WriteLine("Please provide a filter for the methods you want to run. This can be either the name of the method or its namespace (after 'BH.Test').");
```